sfputil/main.py

```diff
@@ -678,148 +681,281 @@ def eeprom(port, dump_dom, namespace):
+    Returns:
+        tuple(0, dump string) if success else tuple(error_code, error_message)
+    """
+    if target_page is not None:
+        lines = [f'EEPROM hexdump for port {logical_port_name} page {target_page}h']
+    else:
+        lines = [f'EEPROM hexdump for port {logical_port_name}']
+    physical_port = logical_port_to_physical_port_index(logical_port_name)
+    api = platform_chassis.get_sfp(physical_port).get_xcvr_api()
+    is_flat_memory = api.is_flat_memory()
+    for page in pages:
+        if page == 0:
+            lines.append(f'{EEPROM_DUMP_INDENT}A0h dump')
+            if not is_flat_memory:
+                return_code, output = eeprom_dump_general(physical_port, page, 0, SFF8472_A0_SIZE, 0)
+            else:
+                return_code, output = eeprom_dump_general(physical_port, page, 0, PAGE_SIZE, 0)
+            if return_code != 0:
+                return return_code, 'Error: Failed to read EEPROM for A0h!'
+            lines.append(output)
+        elif page == 1:
+            lines.append(f'\n{EEPROM_DUMP_INDENT}A2h dump (lower 128 bytes)')
+            return_code, output = eeprom_dump_general(physical_port, page, SFF8472_A0_SIZE, PAGE_SIZE, 0)
+            if return_code != 0:
+                return ERROR_NOT_IMPLEMENTED, 'Error: Failed to read EEPROM for A2h!'
+            lines.append(output)
         else:
-            output += '\n\n{}Upper page {}h'.format(indent, page)
-            output += hexdump(indent, page_dump, PAGE_OFFSET)
-    except NotImplementedError:
-        click.echo("Sfp.read_eeprom() is currently not implemented for this platform")
-        sys.exit(ERROR_NOT_IMPLEMENTED)
-    except ValueError:
-        click.echo("Please enter a numeric page number")
-        sys.exit(ERROR_NOT_IMPLEMENTED)
+            lines.append(f'\n{EEPROM_DUMP_INDENT}A2h dump (upper 128 bytes) page {page - 2:x}h')
+            return_code, output = eeprom_dump_general(physical_port, page, SFF8472_A0_SIZE + PAGE_OFFSET + page * PAGE_SIZE, PAGE_SIZE, PAGE_SIZE)
+            if return_code != 0:
+                return ERROR_NOT_IMPLEMENTED, 'Error: Failed to read EEPROM for A2h upper page!'
+            lines.append(output)
+
+    lines.append('') # add a new line
+    return 0, '\n'.join(lines)
+
+
+def eeprom_dump_general(physical_port, page, overall_offset, size, page_offset, no_format=False):
+    """
+    Dump module EEPROM.
+    Args:
+        physical_port: physical port index
+        page: module EEPROM page number
+        overall_offset: overall offset in flat memory
+        size: size of bytes to be dumped
+        page_offset: offset within a page, only for print purpose
+        no_format: False if dump with hex format else dump with flat hex string. Default False.
+
+    Returns:
+        tuple(0, dump string) if success else tuple(error_code, error_message)
+    """
+    sfp = platform_chassis.get_sfp(physical_port)
+    page_dump = sfp.read_eeprom(overall_offset, size)
```

Comment:
@Junchao-Mellanox just a suggestion, if you could rename overall_offset -> flat_offset that will reflect this offset is considered when viewing the EEPROM a single contigious FLAT memeory